conjure-java-jackson-serialization/src/main/java/com/palantir/conjure/java/serialization/InstrumentedJsonFactory.java

```diff
@@ -0,0 +1,710 @@
+        super(builder);
+        this.parsedStringLength = parsedStringLength;
+        this.creationStackTrace = creationStackTrace;
+    }
+
+    private InstrumentedJsonFactory(
+            JsonFactory src, @Nullable ObjectCodec codec, Histogram parsedStringLength, Throwable creationStackTrace) {
+        super(src, codec);
+        this.parsedStringLength = parsedStringLength;
+        this.creationStackTrace = creationStackTrace;
+    }
+
+    @Override
+    public TSFBuilder<?, ?> rebuild() {
+        return new JsonFactoryBuilder(this) {
+            @Override
+            public JsonFactory build() {
+                return new InstrumentedJsonFactory(this, parsedStringLength, creationStackTrace);
+            }
+        };
+    }
+
+    @Override
+    public JsonFactory copy() {
+        return new InstrumentedJsonFactory(this, null, parsedStringLength, creationStackTrace);
+    }
+
+    @Override
+    protected JsonParser _createParser(InputStream in, IOContext ctxt) throws IOException {
+        return wrap(super._createParser(in, ctxt));
+    }
+
+    @Override
+    protected JsonParser _createParser(Reader reader, IOContext ctxt) throws IOException {
+        return wrap(super._createParser(reader, ctxt));
+    }
+
+    @Override
+    protected JsonParser _createParser(char[] data, int offset, int len, IOContext ctxt, boolean recyclable)
+            throws IOException {
+        return wrap(super._createParser(data, offset, len, ctxt, recyclable));
+    }
+
+    @Override
+    protected JsonParser _createParser(byte[] data, int offset, int len, IOContext ctxt) throws IOException {
+        return wrap(super._createParser(data, offset, len, ctxt));
+    }
+
+    @Override
+    protected JsonParser _createParser(DataInput input, IOContext ctxt) throws IOException {
+        return wrap(super._createParser(input, ctxt));
+    }
+
+    private JsonParser wrap(JsonParser input) {
+        if (input == null || input instanceof InstrumentedJsonParser) {
+            return input;
+        }
+        return new InstrumentedJsonParser(input, parsedStringLength, creationStackTrace);
+    }
+
+    private static final class InstrumentedJsonParser extends JsonParser {
+        private final JsonParser delegate;
+        private final Histogram parsedStringLength;
+        private final Throwable factoryCreationStackTrace;
+
+        InstrumentedJsonParser(JsonParser delegate, Histogram parsedStringLength, Throwable factoryCreationStackTrace) {
+            this.delegate = delegate;
+            this.parsedStringLength = parsedStringLength;
+            this.factoryCreationStackTrace = factoryCreationStackTrace;
+        }
+
+        private String recordStringLength(String value) {
+            if (value != null) {
+                int length = value.length();
+                parsedStringLength.update(length);
+                if (length > 1_000_000 && LOGGING_RATE_LIMITER.tryAcquire()) {
+                    log.info(
```

Comment:
warn?

Reply:
I could go either way, initially I want to understand the state of the world before I ask teams to investigate themselves, but it's unlikely folks will take action based on a new `log.warn` any more than a `log.info` until we add monitors.